drapergem/draper
# ApplicationDecorator generator

## Description
This branch introduces an installation generator.  Running `rails generate draper:install` will create an `ApplicationDecorator` class, if none already exists.  The purpose of this change is encourage use of the Rails' `Application` naming convention by creating a parent decorator class from which others extend.  Due to [existing work](https://github.com/drapergem/draper/blob/master/lib/generators/rails/decorator_generator.rb#L17), generated decorators will inherit from `ApplicationDecorator`, if present.  More information on this feature request can be found [here](https://github.com/drapergem/draper/issues/727).

## Testing
1. Run `rails generate draper:install` in the dummy app.
1. Ensure the class `ApplicationDecorator` is created in `app/decorators/application_decorator.rb`.
1. Ensure newly-generated decorators extend from `ApplicationDecorator` when present.

## To-Dos
N/A

## References
* [GitHub Issue 727](https://github.com/drapergem/draper/issues/727)
